mcs/class/test-helpers/PathHelpers.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace MonoTests.Helpers {
+
+	public static class PathHelpers
```

Comment:
I'd prefer using [CoreFX TempDirectory.cs](https://github.com/dotnet/corefx/blob/24db8490eb6bdeeb2c4353d5b7f9b786546cfa2b/src/Common/tests/System/IO/TempDirectory.cs) instead (either directly or if you really care about including the assembly name then copy the source and adapt it).

Reply:
Is it the IDisposable pattern of TempDirectory you prefer?

That would require more changes to the tests.

I'd also have to copy & adapt the source, since operates on a "hope for no duplicate directories" approach (it's not parallel safe).

Reply:
Please consider just generating a guid.  Much less code that way. No need to check for  existance, etc. It is what they were meant for.

Reply:
@rolfbjarne the IDisposable and not doing all the special magic code that Jay seems to dislike too. Alternatively use `Path.GetRandomFileName()` (it just returns a unique path despite the name) instead of hand rolling the implementation.

Why does the IDisposable pattern need changes to the tests?

Reply:
Very random does sound promising/adequate/simple but I still like guids.

Reply:
> Why does the IDisposable pattern need changes to the tests?

Using the IDisposable pattern means disposing the TempDirectories, which means the test code needs to be changed to dispose them (adding `using`s, which probably means code will end up re-indented, which means a lot of whitespace changes, etc - nothing groundbreaking, but the diff would end up bigger than what it's today).

Reply:
@rolfbjarne you don't need to explicitly dispose the temp directory, it'll be GC'd once it's no longer used.